src/thread.rs

```diff
@@ -183,7 +183,7 @@ struct JoinState<T> {
     _marker: PhantomData<T>,
 }
 
-impl<T: Send> JoinState<T> {
+impl<T> JoinState<T> {
```

Comment:
We don't really need this bound because `JoinState` cannot be created unless `T: Send`. Note that [`JoinHandle`](https://doc.rust-lang.org/std/thread/struct.JoinHandle.html) doesn't have this impl bound either.